### PR TITLE
CVE-2004-0942

### PR DESCRIPTION
**Pull request by TheGuywithTheHat**
Filled out CVE-2004-0942

### CODE REVIEW

File: cves/CVE-2004-0942.yml [hunk 1]
```diff
@@ -31,7 +31,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  If a request is sent that contains tons of whitespace in an HTTP header, the
+  server consumes an excessive amount of CPU or memory (various sources
```
Comment (Anorwen) [inline cves/CVE-2004-0942.yml:36]: What sources said the vulnerability allowed for excessive memory consumption? I'm having trouble finding the other sources.

File: cves/CVE-2004-0942.yml [hunk 2]
```diff
@@ -49,6 +52,8 @@ fixes_vcc_instructions: |
 fixes:
    - commit: f2ef1c9ece9deeb8cc22a7e7ab1415374d9283f2
      note:
+   - commit:
```
Comment (Anorwen) [inline cves/CVE-2004-0942.yml]: Does not having this break the CI pipeline? Otherwise there is no point in adding them.

Comment (Anorwen) [inline cves/CVE-2004-0942.yml]:
```suggestion
```

File: cves/CVE-2004-0942.yml [hunk 3]
```diff
@@ -49,6 +52,8 @@ fixes_vcc_instructions: |
...
+     note:
```
Comment (Anorwen) [inline cves/CVE-2004-0942.yml]:
```suggestion
```

File: cves/CVE-2004-0942.yml [hunk 4]
```diff
@@ -118,8 +127,11 @@ interesting_commits:
     interesting in light of the lessons learned from this vulnerability. Any
     emerging themes?
   commits:
-    - commit:
-      note:
+    - commit: 3d5263c6dcdf517a9eb5a0ab916e1ad49d354355
+      note: |
+        Fix for CVE-2004-0493, which is an extremely similar vulnerability. It
```
Comment (Anorwen) [inline cves/CVE-2004-0942.yml]: In the last part of this document it mentions that this commit was one of their mistakes, why is that? Did this introduce a vulnerability or was the vulnerability patched into a new one?

**Comment by Anorwen** [comment]
@TheGuywithTheHat I would give this 4 points because it's interesting how they patched the file, and a very identical vulnerability appeared again because they weren't thorough enough.

File: cves/CVE-2004-0942.yml (hunk 1 above)
Comment (TheGuywithTheHat) [reply cves/CVE-2004-0942.yml:36]: the fix commit: https://github.com/apache/httpd/commit/f2ef1c9ece9deeb8cc22a7e7ab1415374d9283f2

File: cves/CVE-2004-0942.yml (hunk 4 above)
Comment (TheGuywithTheHat) [reply cves/CVE-2004-0942.yml]: Does my latest commit clear it up?

**Pull request merged**